tests/epyccel/test_epyccel_lists.py

```diff
@@ -118,6 +118,74 @@ def f():
     epyc_f = epyccel(f, language=language)
     assert f() == epyc_f()
 
+def test_insert_basic(language):
+    def f():
+        a = [1, 2, 3]
+        a.insert(4, 4)
+        return a
+
+    epyc_f = epyccel(f, language=language)
+    assert f() == epyc_f()
+
+def test_insert_multiple(language):
+    def f():
+        a = [1, 2, 3]
+        a.insert(4, 4)
```

Comment:
@yguclu this is apparently valid code in Python:
```
a = [1,2,3]
a.insert(100, 4)
```
We can usea an insert index much larger than the size of the list. The element is then simply appended. Is this something we want to support? From a performance perspective it seems similar to negative indexing

Reply:
Yes, I think we should support it. Lists cannot compete with arrays performance-wise, and list insertion is definitely not a fast operation. Most likely, comparing the given index with the size of `a` has a totally negligible overhead.

Reply:
@Farouk-Echaref I suppose that the Python printer has no issues supporting this syntax. So I suggest to add a test for it.

Reply:
The example above does test this. You can insert at positions `0, 1, 2, 3` or out of bounds (4)

Reply:
I think inserting at the index 4 is already is out of bounds, supporting this in the python printer is no issue, my concerns were the equivalent generated code for C and Fortran